modules/linear_boltzmann_solvers/b_discrete_ordinates_solver/sweep_chunks/aah_sweep_chunk.cc

```diff
@@ -39,277 +37,242 @@ AAH_SweepChunk::Sweep(AngleSet& angle_set)
+                                       IsSurfaceSourceActive());
+
+            if (not psi)
               continue;
 
-            for (int gsg = 0; gsg < gs_ss_size_; ++gsg)
-              b_[gsg][i] += psi[gsg] * mu_Nij;
+            for (int gsg = 0; gsg < gs_ss_size; ++gsg)
+              b[gsg][i] += psi[gsg] * mu_Nij;
           } // for face node j
         }   // for face node i
       }     // for f
 
       // Looping over groups, assembling mass terms
-      for (int gsg = 0; gsg < gs_ss_size_; ++gsg)
+      for (int gsg = 0; gsg < gs_ss_size; ++gsg)
       {
-        g_ = gs_gi_ + gsg;
-        gsg_ = gsg;
-        sigma_tg_ = sigma_t[g_];
-
-        const auto& M = *M_;
-        const auto& m2d_op = groupset_.quadrature_->GetMomentToDiscreteOperator();
+        double sigma_tg = sigma_t[gs_gi + gsg];
 
         // Contribute source moments q = M_n^T * q_moms
-        for (int i = 0; i < cell_num_nodes_; ++i)
+        for (int i = 0; i < cell_num_nodes; ++i)
         {
           double temp_src = 0.0;
           for (int m = 0; m < num_moments_; ++m)
           {
-            const size_t ir = cell_transport_view_->MapDOF(i, m, static_cast<int>(g_));
-            temp_src += m2d_op[m][direction_num_] * q_moments_[ir];
-          } // for m
-          source_[i] = temp_src;
-        } // for i
-
-        // Mass Matrix and Source
-        // Atemp  = Amat + sigma_tgr * M
-        // b     += M * q
-        for (int i = 0; i < cell_num_nodes_; ++i)
+            const size_t ir = cell_transport_view.MapDOF(i, m, static_cast<int>(gs_gi + gsg));
+            temp_src += m2d_op[m][direction_num] * source_moments_[ir];
+          }
+          source[i] = temp_src;
+        }
+
+        // Mass matrix and source
+        // Atemp = Amat + sigma_tgr * M
+        // b += M * q
+        for (int i = 0; i < cell_num_nodes; ++i)
         {
           double temp = 0.0;
-          for (int j = 0; j < cell_num_nodes_; ++j)
+          for (int j = 0; j < cell_num_nodes; ++j)
           {
             const double Mij = M[i][j];
-            Atemp_[i][j] = Amat_[i][j] + Mij * sigma_tg_;
-            temp += Mij * source_[j];
-          } // for j
-          b_[gsg_][i] += temp;
-        } // for i
+            Atemp[i][j] = Amat[i][j] + Mij * sigma_tg;
+            temp += Mij * source[j];
+          }
+          b[gsg][i] += temp;
+        }
 
         // Solve system
-        GaussElimination(Atemp_, b_[gsg], static_cast<int>(cell_num_nodes_));
-      }
+        GaussElimination(Atemp, b[gsg], static_cast<int>(cell_num_nodes));
+      } // for gsg
 
-      // Flux updates
-      const auto& d2m_op = groupset_.quadrature_->GetDiscreteToMomentOperator();
+      // Update phi
       auto& output_phi = GetDestinationPhi();
-
       for (int m = 0; m < num_moments_; ++m)
       {
-        const double wn_d2m = d2m_op[m][direction_num_];
-        for (int i = 0; i < cell_num_nodes_; ++i)
+        const double wn_d2m = d2m_op[m][direction_num];
```

Comment:
here, the meaning of the indices is correct. as you can see, it's identical to m2d while the meaning should be swapped for m2d, unless there is an efficiency reason.